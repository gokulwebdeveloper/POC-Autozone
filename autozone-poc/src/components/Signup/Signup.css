.signuphead{
    padding: 1rem;
}

.signupcontainer{
    border: solid 1px #d0d0dc;
    border-radius: 20px;
    margin-top: 1%;
    margin-bottom: 1%;
}
.btn-custome{
    border-radius: 10px;
}