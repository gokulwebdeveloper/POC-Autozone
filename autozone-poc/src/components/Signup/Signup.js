import React, { useState } from "react";
//import {useForm} from 'react-hook-form';
import axios from "axios";
import InputWithLabel from "../../utlis/formComponents/InputWithLabel";
import "./Signup.css";
import Button from "../../utlis/button/Button";

//export default class Signin extends Component{

  export default function Signup () {
  // const {register, handleSubmit, errors, watch}=useForm();
  const [firstname, setFirstName] = useState("");
  const [lastname, setLastName] = useState("");
  const [emailid, setEmailid] = useState("");
  const [password, setPasswords] = useState("");
  const [confirmpassword, setConfirmPassword] = useState("");

  const onSubmit = (evt) => {
    const data = {};
    evt.preventDefault();
    data.firstname = firstname;
    data.lastname = lastname;
    data.emailid = emailid;
    data.password = password;
    axios.post("http://localhost:5000/user/signup", data).then((response) => {
      console.log(response);
    });
  };
  return (
    <div>
      <div className="container h-100">
        <div className="row h-100 align-items-center">
          <div className="col-md-4"></div>
          <div className="col-md-4 align-items-center signupcontainer">
            <div className="page-title text-center">
              <h2 className="signuphead">Sign Up</h2>

              <form onSubmit={onSubmit}>
                    
              <div className="col-md-12 mb-4">    
                  <InputWithLabel
                        labelText="First Name"
                        inputId="first_Name"
                        inputType="text"
                        onChange={(e) => setFirstName(e.target.value)}
                        isRequired={true}/>
                        </div>
              <div className="col-md-12 mb-4">
                  <InputWithLabel
                        labelText="Last Name"
                        inputId="last_Name"
                        inputType="text"
                        onChange={(e) => setLastName(e.target.value)}
                        isRequired={true}/>
                        </div>
                        <div className="col-md-12 mb-4">
                  <InputWithLabel
                        labelText="Email Address"
                        inputId="email_Address"
                        inputType="email"
                        onChange={(e) => setEmailid(e.target.value)}
                        isRequired={true} />
                        </div>
                        <div className="col-md-12 mb-4">
                  <InputWithLabel
                        labelText="Password"
                        inputId="passwords"
                        type="password"
                        onChange={(e) => setPasswords(e.target.value)}
                        isRequired={true}/>
                        </div>
                        <div className="col-md-12 mb-4">
                  <InputWithLabel
                        labelText="Confirm Password"
                        inputId="confirm_password"
                        type="password"
                        onChange={(e) => setConfirmPassword(e.target.value)}
                        isRequired={true}/>
                        </div>
                <br />
                <br />
                <Button btnText="Signup" type="submit" />
                <br />
              </form>
              <br />

              <br />
              <br />
            </div>
          </div>
          <div className="col-md-4"></div>
        </div>
      </div>
    </div>
  );
};

