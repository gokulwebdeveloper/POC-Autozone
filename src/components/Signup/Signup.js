import React, { useState } from "react";
import $ from "jquery";
import axios from "axios";
import InputWithLabel from "../../utlis/formComponents/InputWithLabel";
import "./Signup.css";
import Button from "../../utlis/button/Button";

//export default class Signin extends Component{

export default function Signup(props) {
  // const {register, handleSubmit, errors, watch}=useForm();
  const [firstname, setFirstName] = useState("");
  const [lastname, setLastName] = useState("");
  const [emailid, setEmailid] = useState("");
  const [password, setPasswords] = useState("");
  const [confirmpassword, setConfirmPassword] = useState("");
  const [msgerror, setmsgerror] = useState(false);

  $(document).ready(function () {
    $("#confirm_password").on("blur", function () {
      $("input[required]").each(function (element) {
        let val = $(this).val();
        if (val !== password) {
          $(`.confirm_password_test .formErrorMsg`).text(
            "please provide same password for confirm password"
          );
          $(`.confirm_password_test .formErrorMsg`).show();
        } else {
          $(`.confirm_password_test .formErrorMsg`).hide();
        }
      });
    });
  });

  const onSubmit = (evt) => {
    const data = {};
    evt.preventDefault();
    data.firstname = firstname;
    data.lastname = lastname;
    data.emailid = emailid;
    data.password = password;
    axios
      .post("http://localhost:5000/user/signup", data)
      .then((response) => {
        window.location.assign("/signin");
      })
      .catch((response) => {
        data.msgerror = true;
        window.scrollTo(0, 0);
        setmsgerror(data.msgerror);
      });
  };
  return (
    <div>
      <div className="container-fluid h-100 bgimage">
      {msgerror ? (
        <div className="row justify-content-md-center">
          <div className="col-md-12 message"></div>
          <div className="alert alert-danger col-md-4" role="alert">
            invalid Email id or password!
          </div>
        </div>
      ) : null}
        <div className="row h-100 align-items-center">
          <div className="col-md-5">
            <h1 className="signupExplanation">
              AutoZone has over 5,500 locations across the United States where
              we provide great customer service at every store every day! Become
              a member and get even more great benefits
            </h1>
          </div>
          <div className="col-md-4 signupcontainer">
            <div className="page-title">
              <h2 className="signuphead">Sign Up</h2>

              <form onSubmit={onSubmit}>
                <div className="row namefeild">
                  <div className="col-md-6">
                    <InputWithLabel
                      labelText="First Name"
                      inputId="first_Name"
                      inputType="text"
                      onChange={(e) => setFirstName(e.target.value)}
                      isRequired={true}
                    />
                  </div>

                  <div className="col-md-6">
                    {" "}
                    <InputWithLabel
                      labelText="Last Name"
                      inputId="last_Name"
                      inputType="text"
                      onChange={(e) => setLastName(e.target.value)}
                      isRequired={true}
                    />
                  </div>
                </div>

                <div className="col-md-12 mb-4">
                  <InputWithLabel
                    labelText="Email Address"
                    inputId="email_Address"
                    inputType="email"
                    onChange={(e) => setEmailid(e.target.value)}
                    isRequired={true}
                  />
                </div>
                <div className="col-md-12 mb-4">
                  <InputWithLabel
                    labelText="Password"
                    inputId="passwords"
                    inputType="password"
                    onChange={(e) => setPasswords(e.target.value)}
                    isRequired={true}
                  />
                </div>
                <div className="col-md-12 mb-4 confirm_password_test">
                  <InputWithLabel
                    labelText="Confirm Password"
                    inputId="confirm_password"
                    inputType="password"
                    onChange={(e) => setConfirmPassword(e.target.value)}
                    isRequired={true}
                  />
                </div>
                <br />
                <div className="btnpostion">
                  {" "}
                  <Button btnText="Signup" type="submit" />
                </div>
                <br />
              </form>
              <br />

              <br />
              <br />
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}
