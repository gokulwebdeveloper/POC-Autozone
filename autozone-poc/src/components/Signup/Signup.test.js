import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import Signup from "./Signup";

let page;
//mount component for testing
beforeEach(() => {
  page = mount( <Signup />);
});

//Cleanup after test execution
afterEach(() => {
  page.unmount();
});

describe("test Signup Page", () => {
  //test if Signup component exists
  it("Signup page Exists", () => {
    expect(page.exists()).toBeTruthy();
  });

  //if it has title as "Sign in"
  it('has titles as "Sign Up"', () => {
    expect(page.find("h2").first().text()).toBe("Sign Up");
  });


  //if it has 1 Button componenent
  it("has 1 Button componenent", () => {
    expect(page.find("Button")).toHaveLength(1);
  });

  //if Button has text Signup
  it("Button has text Signup", () => {
    expect(page.find("Button").text()).toBe("Signup");
  });

  //if it has 1 form for Signup
  it("has 1 form for Signup", () => {
    expect(page.find("form")).toHaveLength(1);
  });

  //SnapShot Match test
  it("SnapShot Match", () => {
    const page = renderer.create(<Signup />);
    expect(page).toMatchSnapshot();
  });
});
