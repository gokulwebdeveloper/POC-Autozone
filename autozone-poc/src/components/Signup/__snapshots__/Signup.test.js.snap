// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Signup Page SnapShot Match 1`] = `
<div>
  <div
    className="container h-100"
  >
    <div
      className="row h-100 align-items-center"
    >
      <div
        className="col-md-4"
      />
      <div
        className="col-md-4 align-items-center signupcontainer"
      >
        <div
          className="page-title text-center"
        >
          <h2
            className="signuphead"
          >
            Sign Up
          </h2>
          <form
            onSubmit={[Function]}
          >
            <div
              className="col-md-12 mb-4"
            >
              <label
                htmlFor="first_Name"
              >
                First Name
                 
                <span>
                  *
                </span>
              </label>
              <input
                className="form-control"
                defaultValue=""
                id="first_Name"
                onBlur={[Function]}
                onChange={[Function]}
                required={true}
                type="text"
              />
              <p
                className="formErrorMsg"
              />
            </div>
            <div
              className="col-md-12 mb-4"
            >
              <label
                htmlFor="last_Name"
              >
                Last Name
                 
                <span>
                  *
                </span>
              </label>
              <input
                className="form-control"
                defaultValue=""
                id="last_Name"
                onBlur={[Function]}
                onChange={[Function]}
                required={true}
                type="text"
              />
              <p
                className="formErrorMsg"
              />
            </div>
            <div
              className="col-md-12 mb-4"
            >
              <label
                htmlFor="email_Address"
              >
                Email Address
                 
                <span>
                  *
                </span>
              </label>
              <input
                className="form-control"
                defaultValue=""
                id="email_Address"
                onBlur={[Function]}
                onChange={[Function]}
                required={true}
                type="email"
              />
              <p
                className="formErrorMsg"
              />
            </div>
            <div
              className="col-md-12 mb-4"
            >
              <label
                htmlFor="passwords"
              >
                Password
                 
                <span>
                  *
                </span>
              </label>
              <input
                className="form-control"
                defaultValue=""
                id="passwords"
                onBlur={[Function]}
                onChange={[Function]}
                required={true}
                type="password"
              />
              <p
                className="formErrorMsg"
              />
            </div>
            <div
              className="col-md-12 mb-4"
            >
              <label
                htmlFor="confirm_password"
              >
                Confirm Password
                 
                <span>
                  *
                </span>
              </label>
              <input
                className="form-control"
                defaultValue=""
                id="confirm_password"
                onBlur={[Function]}
                onChange={[Function]}
                required={true}
                type="password"
              />
              <p
                className="formErrorMsg"
              />
            </div>
            <br />
            <br />
            <button
              className="btn essence-btn"
              type="submit"
            >
              Signup
            </button>
            <br />
          </form>
          <br />
          <br />
          <br />
        </div>
      </div>
      <div
        className="col-md-4"
      />
    </div>
  </div>
</div>
`;
